apps/site/assets/ts/components/__tests__/SearchBoxTest.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { ReactWrapper, mount } from "enzyme";
+import SearchBox from "../SearchBox";
+
+describe("SearchBox", () => {
+  let wrapper: ReactWrapper;
+  let spy: jest.Mock;
+  beforeEach(() => {
+    spy = jest.fn();
+    wrapper = mount(
+      <SearchBox
+        id="test-search"
+        labelText="Search for a thing"
+        onChange={spy}
+      />
+    );
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+  });
+
+  it("calls onChange function on change event", () => {
+    wrapper
+      .find("input#test-search")
+      .simulate("change", { target: { value: "foo" } });
+    wrapper.setProps({}); // updates useEffect
```

Comment:
Yeah, its more of a workaround because functional React components don't have state or instances that Enzyme's methods can test well. Additionally, `wrapper.update()` doesn't work for this, [though you'd think it would](https://github.com/airbnb/enzyme/issues/2254). 

As far as I can tell, **_best practices_** around testing functional React components, particularly those using Hooks, have not really coalesced yet... and what little wisdom exists is spread across a wonderfully varied set of libraries and helpers.

That said. your mention of `act()` made me realize that implementing some of these tests with `react-test-renderer` works, and may look slightly less hacky! 

For example this:
```tsx
it("renders properly if multiple matches", () => {
    const wrapper = mount(
      <MatchHighlight
        text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
        query="and"
      />
    );
    wrapper.setProps({}); // actually updates the wrapper post-useEffect
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".u-highlight")).toHaveLength(1);
  });
```
becomes this:
```tsx
  it("highlights only the first match if there are multiple matches", () => {
    act(() => {
      component = create(
        <MatchHighlight
          text="As part of our $8 billion, 5-year capital investment plan, we're renovating stations, modernizing fare collection systems, upgrading services for our buses, subways, and ferries, and improving the accessibility of the entire system."
          query="and"
        />
      );
    });

    const instance = component!.root;
    const highlight = instance.findByType("b");
    expect(highlight.children).toHaveLength(1);
  });
```

But would still necessitate additional/alternate tools to handle changing inputs and whatnot.

Reply:
> a wonderfully varied set of libraries and helpers

So far, so JavaScript 😓 Makes me wonder how exactly Facebook is testing all their own functional components...

I can appreciate the slightly-less-hacky-ness of the `act` approach, though my personal preference would be to keep all our tests (or as many as possible) aligned on the same framework. This workaround is officially documented by Enzyme, and they've left the issue open which makes me hopeful it will have a "real" resolution eventually. Nice work sleuthing out that information 😃